### PR TITLE
Fix propagating of default date for milestoned properties from association

### LINKED ISSUE #1422: Bug: Unable to propagate default dates for milestoned properties from association
### Similar issues

- [X] I have searched and found no existing similar issues

### How are you using Studio?

Legend Studio

### Current and expected behavior

Unable to propagate default dates for milestoned properties from association

### Steps to reproduce

Copy and paste below model data and hit `Query...` for the class `Firm` and drag and drop property `employees/name`. You don't see date getting propagated when you toggle to text mode.

### Model data

```text
###Relational
Database my::db
(
  Table PersonTable
  (
    NAME CHAR(200),
    FIRMID INTEGER
  )
  Table FirmTable
  (
    FIRMNAME CHAR(200),
    ID INTEGER
  )

  Join Firm_Person(PersonTable.FIRMID = FirmTable.ID)
)


###Pure
Class my::Firm
{
  legalName: String[1];
  id: Integer[1];
}

Class <<meta::pure::profiles::temporal.businesstemporal>> my::Person
{
  name: String[1];
  firmID: Integer[1];
}

Association my::Employment
{
  firm: my::Firm[*];
  employees: my::Person[*];
}


###Mapping
Mapping my::map
(
  my::Person[per1]: Relational
  {
    ~mainTable [my::db]PersonTable
    name: [my::db]PersonTable.NAME
  }
  my::Firm[fir1]: Relational
  {
    ~mainTable [my::db]FirmTable
    legalName: [my::db]FirmTable.FIRMNAME
  }

  my::Employment: Relational
  {
    AssociationMapping
    (
      firm[per1,fir1]: [my::db]@Firm_Person,
      employees[fir1,per1]: [my::db]@Firm_Person
    )
  }
)
```

### Environment

```text
Windows
```

### Possible solution and workaround

_No response_

### Contribution

- [X] I would like to work on the fix for this issue

### PR DESCRIPTION
**Pull request by gayathrir11**
## Summary

Fixes #1422 

## How did you test this change?

- [ ] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

https://user-images.githubusercontent.com/87973126/185099547-58f7e4b1-f847-48be-84a8-85a37e31a9fd.mp4

### CODE REVIEW

File: packages/legend-application-query/src/stores/QueryBuilderPropertyEditorState.ts [hunk 1]
```diff
@@ -67,9 +67,14 @@ export const getDerivedPropertyMilestoningSteoreotype = (
   if (owner instanceof Class) {
     return getMilestoneTemporalStereotype(owner, graph);
   } else if (owner instanceof Association) {
-    if (owner._generatedMilestonedProperties.length) {
-      const ownerClass =
-        owner._generatedMilestonedProperties[0]?.genericType.value.rawType;
+    if (
+      owner._generatedMilestonedProperties.filter(
+        (prop) => !prop.name.startsWith(property.name),
```
Comment (akphi) [inline packages/legend-application-query/src/stores/QueryBuilderPropertyEditorState.ts:72]: don't think this is a good idea to check for matching properties.

File: .changeset/strange-pumas-cheat.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@finos/legend-application-query": patch
+---
+
+Fix propagating of default date for milestoned properties from association([#1422](https://github.com/finos/legend-studio/issues/1422)).
```
Comment (akphi) [inline .changeset/strange-pumas-cheat.md:5]:
```suggestion
Fix propagating of default date for milestoned properties from association ([#1422](https://github.com/finos/legend-studio/issues/1422)).
```

**Review by akphi — CHANGES_REQUESTED** [review]

File: packages/legend-application-query/src/stores/QueryBuilderPropertyEditorState.ts [hunk 3]
```diff
@@ -67,9 +68,18 @@ export const getDerivedPropertyMilestoningSteoreotype = (
   if (owner instanceof Class) {
     return getMilestoneTemporalStereotype(owner, graph);
   } else if (owner instanceof Association) {
-    if (owner._generatedMilestonedProperties.length) {
+    const generatedMilestonedProeprties =
+      owner._generatedMilestonedProperties.filter(
+        (prop) =>
+          prop.name !== property.name &&
```
Comment (akphi) [inline packages/legend-application-query/src/stores/QueryBuilderPropertyEditorState.ts:74]: I think its fair at this point to make this a helper function in DomainHelper. What do you think?

File: packages/legend-application-query/src/stores/QueryBuilderPropertyEditorState.ts [hunk 4]
```diff
@@ -67,9 +68,18 @@ export const getDerivedPropertyMilestoningSteoreotype = (
...
+          prop.name !==
+            `${property.name}${MILESTONING_VERSION_PROPERTY_SUFFIX.ALL_VERSIONS}` &&
+          prop.name !==
+            `${property.name}${MILESTONING_VERSION_PROPERTY_SUFFIX.ALL_VERSIONS_IN_RANGE}`,
+      );
+    if (generatedMilestonedProeprties.length) {
       const ownerClass =
-        owner._generatedMilestonedProperties[0]?.genericType.value.rawType;
+        generatedMilestonedProeprties[0]?.genericType.value.rawType;
```
Comment (akphi) [inline packages/legend-application-query/src/stores/QueryBuilderPropertyEditorState.ts:82]: Lets fix the typo in properties 😉

**Review by akphi — CHANGES_REQUESTED** [review]

File: packages/legend-application-query/src/stores/QueryBuilderPropertyEditorState.ts (hunk 3 above)
Comment (gayathrir11) [reply packages/legend-application-query/src/stores/QueryBuilderPropertyEditorState.ts:74]: yeah

File: packages/legend-graph/src/graph/helpers/DomainHelper.ts [hunk 5]
```diff
@@ -563,3 +564,19 @@ export const isElementDeprecated = (
         .getProfile(CORE_PURE_PATH.PROFILE_DOC)
         .p_stereotypes.find((s) => s.value === PURE_DEPRECATED_STEREOTYPE),
   );
+
+/**
+ *  Gets generatedMilestonedProperties when the owner of a property is an association
+ */
+export const getGeneratedMilestonedPropertiesForAssociation = (
+  association: Association,
```
Comment (akphi) [inline packages/legend-graph/src/graph/helpers/DomainHelper.ts:572]: --> `propertyOwner: PropertyOwner`

**Review by akphi — CHANGES_REQUESTED** [review]

**Review by akphi — APPROVED** [review]

**Pull request merged**